src/830into.js

```diff
@@ -144,6 +144,64 @@ alasql.into.JSON = function (filename, opts, data, columns, cb) {
 	return res;
 };
 
+// Helper function to convert arrow-notation keys to nested objects
+// e.g., {"details->stock": 20} becomes {"details": {"stock": 20}}
+function unflattenObject(obj) {
```

Comment:
I feel like we are doing dobule work here. Could we just avoid converting them to arrow notation if we are moving them into an object?